src/features/chart/redux/middleware.ts

```diff
@@ -1,8 +1,19 @@
 import { BOOTSTRAP } from "../../../store/constants";
 import { UPDATE_CHART_RANGE } from "./constants";
 import { updateChartDataAction } from "./actions";
+import { Middleware, Dispatch, AnyAction } from "redux";
+import { AppState } from "../../../store";
+import { SocketService } from "../../../services/socketService";
 
-export const chartMiddleware = ({
+export type Dependencies = {
+  socketService: SocketService;
+};
+
+export type ChartMiddleware = (
+  dependencies: Dependencies
+) => Middleware<{}, AppState, Dispatch<AnyAction>>;
```

Comment:
nit: You don't need the 3rd generic parameter here. It is defaulted to `Dispatch`
```suggestion
) => Middleware<{}, AppState>;
```